File: backend/models/device.py
from pydantic import BaseModel


class ExportRequest(BaseModel):
    fileName: str


class RestoreRequest(BaseModel):
    URL: str
    Username: str
    Password: str
    FileSize: int | None = None
    TargetFileName: str | None = None
    CheckSumAlgorithm: str | None = None
    CheckSum: str | None = None


class RestoreExtendedRequest(RestoreRequest):
    CACert: str | None = None
    ClientCert: str | None = None
    PrivateKey: str | None = None

File: frontend/src/app/models/device.ts
export interface ExportRequest {
  fileName: string;
}

export interface RestoreRequest {
  url: string;
  username: string;
  password: string;
  fileSize?: number;
  targetFileName?: string;
  checksumAlgorithm?: string;
  checksum?: string;
}

export interface RestoreExtendedRequest extends RestoreRequest {
  caCert?: string;
  clientCert?: string;
  privateKey?: string;
}
